### PR TITLE
Fix functionList.xml for php

### PR DESCRIPTION
**Pull request by WinterSilence**
Add `# comment` + foreach contruction.

### CODE REVIEW

**Comment by chcg** [comment]
Seems n++ function list changes are still blocked by https://github.com/notepad-plus-plus/notepad-plus-plus/pull/3393 , as no tests are in place yet.

File: PowerEditor/src/functionList.xml [hunk 1]
```diff
@@ -123,21 +123,21 @@
...
 					</function>
 				</classRange>
 				<function
 					mainExpr="^\s*function\s+\w+\s*\("
 				>
 					<functionName>
-						<nameExpr expr="(?!(if|while|for))\w+\s*\(" />
-						<nameExpr expr="(?!(if|while|for))\w+" />
+						<nameExpr expr="(?!(if|while|for|foreach))\w+\s*\(" />
+						<nameExpr expr="(?!(if|while|for|foreach))\w+" />
```
Comment (vlakoff) [inline PowerEditor/src/functionList.xml]: Do you know why `switch` is missing here? Probably an overlook, isn't it?

File: PowerEditor/src/functionList.xml [hunk 2]
```diff
@@ -123,21 +123,21 @@
 						<nameExpr expr="\w+\Z" />
 					</className>
 					<function
-						mainExpr="^\s*((static|public|protected|private|final)*(\s+(static|public|protected|private|final))+\s+)?(function\s+)+(\w+(\s+\w+)?(\s+|\*\s+|\s+\*|\s+\*\s+))?(\w+\s*::)?(?!(if|while|for|switch))[\w~]+\s*\([^\{]*\{"
+						mainExpr="^\s*((static|public|protected|private|final)*(\s+(static|public|protected|private|final))+\s+)?(function\s+)+(\w+(\s+\w+)?(\s+|\*\s+|\s+\*|\s+\*\s+))?(\w+\s*::)?(?!(if|while|for|switch|foreach))[\w~]+\s*\([^\{]*\{"
 					>
 						<functionName>
-							<funcNameExpr expr="(?!(if|while|for|switch))\w+\s*\([^\{]*" />
+							<funcNameExpr expr="(?!(if|while|for|switch|foreach))\w+\s*\([^\{]*" />
```
Comment (vlakoff) [inline PowerEditor/src/functionList.xml]: Should we also add [`do`](https://www.php.net/manual/en/control-structures.do.while.php)?

Comment (vlakoff) [reply PowerEditor/src/functionList.xml]: However, I can't reproduce a false positive happening with these structures.

Could you please provide an example showing the need for these regex negative assertions?

File: PowerEditor/src/functionList.xml [hunk 3]
```diff
@@ -123,21 +123,21 @@
 						<nameExpr expr="\w+\Z" />
 					</className>
 					<function
-						mainExpr="^\s*((static|public|protected|private|final)*(\s+(static|public|protected|private|final))+\s+)?(function\s+)+(\w+(\s+\w+)?(\s+|\*\s+|\s+\*|\s+\*\s+))?(\w+\s*::)?(?!(if|while|for|switch))[\w~]+\s*\([^\{]*\{"
+						mainExpr="^\s*((static|public|protected|private|final)*(\s+(static|public|protected|private|final))+\s+)?(function\s+)+(\w+(\s+\w+)?(\s+|\*\s+|\s+\*|\s+\*\s+))?(\w+\s*::)?(?!(if|while|for|switch|foreach))[\w~]+\s*\([^\{]*\{"
```
Comment (vlakoff) [inline PowerEditor/src/functionList.xml]: (yet) outside the scope of this PR, but I guess the `(\s+|\*\s+|\s+\*|\s+\*\s+)` could be shortened to `(\s+)`, as there can't be such `*` in PHP. Probably the rule has been copied from another language.

Comment (vlakoff) [reply PowerEditor/src/functionList.xml]: Same with the double colon `::`, that can occur in static call but not in function definition.

**Comment by WinterSilence** [comment]
@vlakoff pls, if u want change something send PR

File: PowerEditor/src/functionList.xml (hunk 2 above)
Comment (WinterSilence) [reply PowerEditor/src/functionList.xml]: @vlakoff  function list in `futures` - not all work in curret version. See https://notepad-plus-plus.org/features/function-list.html

**Comment by donho** [comment]
@WinterSilence  Could you provide a sample php code to show your enhancement?

**Comment by sasumner** [comment]
@donho said

>@WinterSilence Could you provide a sample php code to show your enhancement?

An "issue" would have been the perfect place to discuss this and show what a possible enhancement could have done.  But the protocol of "issue first, then PR" seems to have been skipped for this case.

Perhaps priority of PR integration consideration should be given to clearly described issues and well-done PRs.  That way, the people that put the effort in, see the fruits of their work integrated sooner.  :-)

**Comment by donho** [comment]
Just gave some tries, cannot see any point for this PR.
Please provide the samples in an issue and reopen it.

**Pull request closed without merging**